homeassistant/components/amcrest/__init__.py

```diff
@@ -184,6 +200,38 @@ def _wrap_test_online(self, now):
             pass
 
 
+def _monitor_events(hass, name, api, event_codes):
+    event_codes = ",".join(event_codes)
+    start_stop = re.compile(r"Code=([^;]+);action=(Start|Stop)", flags=re.S)
+    while True:
+        try:
+            api.available_flag.wait()
```

Comment:
Can wait() throw an exception here?

Reply:
No. I suppose it could be outside the `try` block.

Reply:
Have you tested unplugging a camera and blocking all IP traffic to the camera to make sure this doesn’t end up in an infinite loop?

Reply:
Yes. When the communication fails, causing the exception, the `api.available_flag` event flag will also be cleared, which will cause it to wait here until the communications start working again (which will result in the flag getting set and the wait will complete.)

Reply:
Actually, to be more accurate, when there are a enough communication errors back to back (whether caused here or elsewhere), the flag will get cleared (see `_start_recovery()` above), which will prevent this thread from starting the event stream again until `_wrap_test_online()` detects the camera is back "on line", at which time the flag will be set and this thread will attempt to continue as usual.

Reply:
Should be ok to resolve by moving the `wait()` out of the try

Reply:
Will do.